### ISSUE TITLE
[Bug]: Appwrite redirect loop

### ISSUE THREAD

**Issue by teknology** [coollabsio/coolify#4332]
### Error Message and Logs

There seems to be a redirect loop preventing Appwrite from loading from the generated url. 

### Steps to Reproduce

1. Install Appwrite from the resource catalog
2. Deploy
3. Try to access the generated link from the link dropdown.
4.

### Coolify Version

v4.0.0-beta.370

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Debian 12

### Additional Information

It doesn't matter of it's a generated url with a private domain or a sslip.io. Both results in a redirect loop.

**Comment by djsisson** [comment]
@teknology untick strip prefix in console and realtime settings

**Comment by teknology** [comment]
That didn't solve the issue. It resulted in Realtime link reporting a 400 bad request and console still has a redirect loop.

**Comment by teknology** [comment]
> [@teknology](https://github.com/teknology) untick strip prefix in console and realtime settings

That didn't solve the issue. It resulted in Realtime link reporting a 400 bad request and console still has a redirect loop.

**Comment by ChristophJeworutzki** [comment]
+1

**Comment by Shadowfita** [comment]
Also experiencing this issue using Caddy. Hopefully resolved soon.

**Comment by alexaka1** [comment]
Just as a sanity check, are you sure you are not hitting this issue: https://coolify.io/docs/knowledge-base/faq#too-many-redirections ?

**Comment by voglster** [comment]
TLDR Same issue, brand new setup clean vps and appwrite only app setup but other apps work fine (dashy)

If anyone wants to debug I am technical enough to help.

I am very new to coolify (came from caprover) so entirely possible that I missed something but the the first app i tried to deploy was Appwrite and hit a redirect loop.  However doing the exact same setup with dashy everything works out of the box.

im guessing something awry with appwrites internal nginx vs traefic and passing path rewrites etc...

curl log of redirect loop (ip/domain changed for obvious reasons)

```
$ curl -I -L --max-redirs 3 -v https://appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com/console                                                               
*   Trying xxx.xxx.xx.xxx:443...
* Connected to appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com (xxx.xxx.xxx.xxx) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
*  start date: Nov 24 03:57:24 2024 GMT
*  expire date: Feb 22 03:57:23 2025 GMT
*  subjectAltName: host "appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com" matched cert's "appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com"
*  issuer: C=US; O=Let's Encrypt; CN=R11
*  SSL certificate verify ok.
* using HTTP/2
* h2h3 [:method: HEAD]
* h2h3 [:path: /console]
* h2h3 [:scheme: https]
* h2h3 [:authority: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x5585e7a58ce0)
> HEAD /console HTTP/2
> Host: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
> user-agent: curl/7.88.1
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
< HTTP/2 301 
HTTP/2 301 
< alt-svc: h3=":443"; ma=2592000
alt-svc: h3=":443"; ma=2592000
< content-type: text/html
content-type: text/html
< date: Sun, 24 Nov 2024 05:00:03 GMT
date: Sun, 24 Nov 2024 05:00:03 GMT
< location: /console
location: /console
< server: nginx/1.25.5
server: nginx/1.25.5
< content-length: 169
content-length: 169

< 
* Connection #0 to host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com left intact
* Issue another request to this URL: 'https://appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com/console'
* Found bundle for host: 0x5585e7a55160 [can multiplex]
* Re-using existing connection #0 with host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
* h2h3 [:method: HEAD]
* h2h3 [:path: /console]
* h2h3 [:scheme: https]
* h2h3 [:authority: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 3 (easy handle 0x5585e7a58ce0)
> HEAD /console HTTP/2
> Host: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
> user-agent: curl/7.88.1
> accept: */*
> 
< HTTP/2 301 
HTTP/2 301 
< alt-svc: h3=":443"; ma=2592000
alt-svc: h3=":443"; ma=2592000
< content-type: text/html
content-type: text/html
< date: Sun, 24 Nov 2024 05:00:04 GMT
date: Sun, 24 Nov 2024 05:00:04 GMT
< location: /console
location: /console
< server: nginx/1.25.5
server: nginx/1.25.5
< content-length: 169
content-length: 169

< 
* Connection #0 to host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com left intact
* Issue another request to this URL: 'https://appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com/console'
* Found bundle for host: 0x5585e7a55160 [can multiplex]
* Re-using existing connection #0 with host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
* h2h3 [:method: HEAD]
* h2h3 [:path: /console]
* h2h3 [:scheme: https]
* h2h3 [:authority: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 5 (easy handle 0x5585e7a58ce0)
> HEAD /console HTTP/2
> Host: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
> user-agent: curl/7.88.1
> accept: */*
> 
< HTTP/2 301 
HTTP/2 301 
< alt-svc: h3=":443"; ma=2592000
alt-svc: h3=":443"; ma=2592000
< content-type: text/html
content-type: text/html
< date: Sun, 24 Nov 2024 05:00:04 GMT
date: Sun, 24 Nov 2024 05:00:04 GMT
< location: /console
location: /console
< server: nginx/1.25.5
server: nginx/1.25.5
< content-length: 169
content-length: 169

< 
* Connection #0 to host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com left intact
* Issue another request to this URL: 'https://appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com/console'
* Found bundle for host: 0x5585e7a55160 [can multiplex]
* Re-using existing connection #0 with host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
* h2h3 [:method: HEAD]
* h2h3 [:path: /console]
* h2h3 [:scheme: https]
* h2h3 [:authority: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 7 (easy handle 0x5585e7a58ce0)
> HEAD /console HTTP/2
> Host: appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com
> user-agent: curl/7.88.1
> accept: */*
> 
< HTTP/2 301 
HTTP/2 301 
< alt-svc: h3=":443"; ma=2592000
alt-svc: h3=":443"; ma=2592000
< content-type: text/html
content-type: text/html
< date: Sun, 24 Nov 2024 05:00:04 GMT
date: Sun, 24 Nov 2024 05:00:04 GMT
< location: /console
location: /console
< server: nginx/1.25.5
server: nginx/1.25.5
< content-length: 169
content-length: 169

< 
* Connection #0 to host appwrite-uk0w0w48sowsowgk0w4kowo0.cld3.example.com left intact
* Maximum (3) redirects followed
curl: (47) Maximum (3) redirects followed
```

**Comment by voglster** [comment]
Unticking strip prefixes does indeed fix it (when unticked for all)

![Image](https://github.com/user-attachments/assets/d5f82817-939d-42c8-a1ef-bc52c0ef3a2c)

**Comment by Shadowfita** [comment]
Unfortunately, I've not had any success with unticking "Strip Prefixes". The issue persists with too many redirects.

**Comment by bizrockman** [comment]
Waited for the update to 370 to see if that error persists. And it does. I set it up with the default template provided with coolify now. Getting the same error of to many redirects.

With a clean setup. Using defaults.

**Comment by djsisson** [comment]
@bizrockman what proxy are you using?

p.s you still need to untick strip prefixes

**Comment by bizrockman** [comment]
> [@bizrockman](https://github.com/bizrockman) what proxy are you using?
> 
> p.s you still need to untick strip prefixes

Hi, I am using Traefik. 

The option strip prefixes is not available. I am using the template that got shipped with coolify. 
Maybe I do not see it. Where should I search?

**Comment by djsisson** [comment]
@bizrockman click settings on realtime and on console

**Comment by bizrockman** [comment]
> [@bizrockman](https://github.com/bizrockman) click settings on realtime and on console

Ok thanks. Both a ticked by default.

**Comment by djsisson** [comment]
@bizrockman it needs to be unticked

**Comment by bizrockman** [comment]
> [@bizrockman](https://github.com/bizrockman) it needs to be unticked

I see. I have unticked the setting. 
Restarted the service.
Cleared browser cache/cookies

Do not untick the main domain of service "Appwrite". 

And now it is running. 

Thanks!! @djsisson

**Comment by Shadowfita** [comment]
Unfortunately, the workarounds above don't seem to work with the caddy proxy.

**Comment by djsisson** [comment]
@Shadowfita caddy proxy i believe has just been fixed, so it should work now in 372?

**Comment by VolenBE** [comment]
Unticking worked for me, I don't know if it's possible for coolify devs to do it, but maybe make sure they are not ticked by default?

**Comment by Shadowfita** [comment]
Can confirm, thank you @djsisson  !

**Comment by LukeAndrewColeman** [comment]
This also worked for me 👍

**Comment by kotatsu-dog** [comment]
This worked for me too
Thank you for giving a solution!

**Comment by kazGuido** [comment]
> Unticking strip prefixes does indeed fix it (when unticked for all 3 web exposed services)
> 
> ![Image](https://github.com/user-attachments/assets/d5f82817-939d-42c8-a1ef-bc52c0ef3a2c)

lifesaver, thx !

**Comment by LukasParke** [comment]
I'm not having success using this template at all, whether or not strip prefixes is checked

**Comment by DavidDeSloovere** [comment]
> I'm not having success using this template at all, whether or not strip prefixes is checked

I also thought it wasn't working. Deleted the whole thing. Created a new one, uncheck "strip prefixes" and then suddenly it worked. 
Tried with another browser. Thinking my main browser was caching the redirect!


You can also curl it, that should not cache the response.

```
curl -IL https://appwrite.yourdomain.here
```


PS: Also added CPU and RAM to my VM.

**Comment by abhishekram404** [comment]
Unticking "Strip Prefixes" for all 3 web exposed services fixed it for me.

**Comment by Bolleks** [comment]
Yes, indeed! The problem is solved this way! Thanks for the help!

**Comment by mr-cozay** [comment]
Thanks you @teknology for sharing your problem, I just launched an appwrite instance on coolify and I encountered the same problem. I followed @kazGuido 's advice, I unchecked the Strip Prefixes box on the 3 web apps (appwrite, console and realtime), and everything worked.

**Comment by andrasbacsai** [comment]
Already fixed.

Strip prefix is auto-disabled for appwrite / console / realtime. Existing services may still need Strip Prefixes unchecked once.
